src/main/scala/is/hail/expr/AnnotationImpex.scala

```diff
@@ -53,6 +53,13 @@ object SparkAnnotationImpex extends AnnotationImpex[DataType, Any] {
         }: _*)
   }
 
+  def annotationImporter(t: Type): (Any) => Annotation = {
```

Comment:
Yes, I did a version of this, but led to a high number of composed, higher-ordered functions (like we had with the AST eval method) and it didn't seem better.  But we could use asm to generic a Type-specific importer, that would be nice.